### PR TITLE
Allow passing `fileName` in the `StoreFileOptions`

### PR DESCRIPTION
**Pull request by amitaibu**

### CODE REVIEW

File: IHP/FileStorage/Types.hs [hunk 1]
```diff
@@ -25,13 +25,15 @@ data StoreFileOptions = StoreFileOptions
     { directory :: Text -- ^ Directory on S3 or inside the @static@ where all files are placed
     , contentDisposition :: Wai.FileInfo LByteString -> IO (Maybe Text) -- ^ The browser uses the content disposition header to detect if the file should be shown inside the browser or should be downloaded as a file attachment. You can provide a function here that returns a custom content-disposition header based on the uploaded file. This currently only works with the S3 storage. See 'contentDispositionAttachmentAndFileName' for standard configuration.
     , preprocess :: Wai.FileInfo LByteString -> IO (Wai.FileInfo LByteString) -- ^ Can be used to preprocess the file before storing it inside the storage. See 'applyImageMagick' for preprocessing images.
+    , fileName :: Maybe UUID -- ^ Optional filename. We use UUID for security measures.
```
Comment (amitaibu) [inline IHP/FileStorage/Types.hs:28]:
```suggestion
    , fileName :: Maybe UUID -- ^ Optional filename. We use UUID for security reasons.
```

**Comment by amitaibu** [comment]
Failing test doesn't seem related

**Comment by amitaibu** [comment]
@mpscholten is ready, but not sure what to do with the failing build

**Comment by mpscholten** [comment]
The error on the build seems to be unrelated.

Can we also add some documentation to the Guide on how to use the new `fileName` attribute? Otherwise it's unlikely people will be aware that this exists now

**Review by mpscholten — APPROVED** [review]
Thanks 👍

**Pull request merged**